### ISSUE TITLE
Does it exist a function to sample irregular points from a chain?

### ISSUE THREAD

**Issue by ErickChacon** [JuliaGeometry/Meshes.jl#182]
Hi, thank you for the effort on *Meshes.jl*. I was considering to use it for a project I am working on and I would appreciate if you can answer the following questions.

- Is there a function to sample points irregularly from a segment? I was able to do it with something like
 ```julia
s = Segment(Point2(0.0, 0.0), Point2(1.0, 1.0))
map(s, rand(10))
```
but is that it the intended approach?
- Is there a function to sample points irregularly from a Chain? I thought the following was going to work, but it does not.
 ```julia
linestring = Chain((0.0, 0.0), (1.0, 1.5), (2.5, 2.0), (3.5, 2.5), (2.0, 0.5))
linestring_sample = sample(linestring, HomogeneousSampling(10))
 ```
Thank you in advance.

**Comment by juliohm** [comment]
The `Segment` supports evaluation at a parameter `t` so you can write `s.(0.0:0.1:1.0)` to get regularly spaced samples.

Regarding irrregularly spaced samples, we first need to add a method for `Segment` similar to this one for `Triangle`:

https://github.com/JuliaGeometry/Meshes.jl/blob/126817df414f80b619a093e72e84e313a96b2d13/src/sampling/homogeneous.jl#L29-L40

Can you submit a PR with this feature? I am happy to review.

**Comment by ErickChacon** [comment]
Great! I will be working on this then. I think it will be something like the following?

```julia
function samplex(rng::AbstractRNG, s::Segment{Dim,T},
                 method::HomogeneousSampling) where {Dim,T}
   A, B = vertices(s)
   function randpoint()
     t = rand(rng, T)
     A + t * (B - A)
   end
   (randpoint() for _ in 1:method.size)
 end
```

**Comment by ErickChacon** [comment]
For the case of sampling from a chain. A homogenous sampling could be achieved by sampling the segments proportionally to their lengths and later the points from the selected segments. Is that the approach you were thinking of? For example the following works.

```julia
using Meshes
using Plots
linest = Chain((0.0, 0.0), (1.0, 1.0), (1.0, 0.0), (2.0, 2.0))
ss = segments(linest) |> collect
ss_samples = sample(1:length(ss), Meshes.Weights(measure.(ss)), 50)
ps_in_chain = [ss[ss_samples[i]](rand()) for i in 1:50] |> collect

plot(linest, lc = :gray)
plot!(ps_in_chain, ms = 2, color = :red, msw = 0)
```

![example](https://user-images.githubusercontent.com/7862458/139155430-98ff8ca1-442f-4632-889e-ae7c158c7f9d.png)

I can try to write a method using the Segment method if that it the desired approach.

**Comment by juliohm** [comment]
Exactly the approach I had in mind. After the first PR is merged, we can then start a new PR that adds a method for Chain. The method could simply do `gset = Collection(collect(segments(chain)))` and call the homogeneous sampling method already defined for geometry sets.

**Comment by ErickChacon** [comment]
> gset = Collection(collect(segments(chain)))

I just tested this and it works as expected! I need to explore more this package!

**Comment by juliohm** [comment]
We can finish that PR and start a new one with this method for Chain. Also, happy to review the other PR with a method in for Segment 👍🏽 :)

**Comment by ErickChacon** [comment]
I probably should close this issue given that the feature has been included with PR #183 and #184.
